crates/sui-e2e-tests/tests/checkpoint_tests.rs

```diff
@@ -35,3 +41,32 @@ async fn basic_checkpoints_integration_test() {
 
     panic!("Did not include transaction in checkpoint in 60 seconds");
 }
+
+#[sim_test]
+async fn checkpoint_split_brain_test() {
+    let count_split_brain_nodes: Arc<Mutex<AtomicUsize>> = Default::default();
+    let count_clone = count_split_brain_nodes.clone();
+    let test_cluster = TestClusterBuilder::new()
+        .with_num_validators(4)
+        .build()
+        .await;
+
+    register_fail_point("split_brain_reached", move || {
+        let counter = count_clone.lock().unwrap();
+        counter.fetch_add(1, Ordering::Relaxed);
+    });
+    // introduce execution nondeterminism for half the committee
+    register_fail_point_if("cp_execution_nondeterminism", || {
+        sui_simulator::current_simnode_id().0 % 2 == 1
+    });
+
+    let tx = make_transfer_sui_transaction(&test_cluster.wallet, None, None).await;
+    test_cluster.execute_transaction(tx).await;
+
+    // provide enough time for
```

Comment:
for ?

Reply:
oh thanks, fixing. We just need enough time for other validators to detect the split brain